packages/open-payments/src/client/token.test.ts

```diff
@@ -0,0 +1,29 @@
+import { createTokenRoutes } from './token'
+import { OpenAPI, HttpMethod, createOpenAPI } from 'openapi'
+import config from '../config'
+import { defaultAxiosInstance, silentLogger } from '../test/helpers'
+
+describe('token', (): void => {
+  let openApi: OpenAPI
+
+  beforeAll(async () => {
+    openApi = await createOpenAPI(config.OPEN_PAYMENTS_AS_OPEN_API_URL)
+  })
+
+  const axiosInstance = defaultAxiosInstance
+  const logger = silentLogger
+  const client = 'https://example.com/.well-known/pay'
+
+  describe('createTokenRoutes', (): void => {
+    test('creates response validator for token requests', async (): Promise<void> => {
+      jest.spyOn(openApi, 'createResponseValidator')
+
+      createTokenRoutes({ axiosInstance, openApi, logger, client })
+      expect(openApi.createResponseValidator).toHaveBeenCalledTimes(1)
+      expect(openApi.createResponseValidator).toHaveBeenCalledWith({
+        path: '/token/{id}',
+        method: HttpMethod.POST
+      })
+    })
+  })
```

Comment:
Add `rotate` tests with  `openApiValidators.successfulValidator`/`openApiValidators.failedValidator`
See:
https://github.com/interledger/rafiki/blob/1124707caf2f6ed9f7ddae777898973f9107b749/packages/open-payments/src/client/outgoing-payment.test.ts#L132

Reply:
Well, to be honest I don't quite understand about this.
In open-payments.test.ts, it's testing this function.

export const getOutgoingPayment = async (
  deps: BaseDeps,
  args: GetArgs,
  validateOpenApiResponse: ResponseValidator<OutgoingPayment>
) => {
}

But in token.ts, it doesn't have that function format, except post() function which will be tested in requests.test.ts

Reply:
The only function in token.ts is rotate() function, which only takes url as parameber not validators like successfulValidators or failed Validators.

Could you give me some more details on this?

Reply:
You would need to move the post() to an exported function (like `getOutgoingPayment`).

Maybe the route specific tests should be having `nock` check the request body, to distinguish from the general `post` tests in `requests.test.ts`
https://github.com/interledger/rafiki/blob/1124707caf2f6ed9f7ddae777898973f9107b749/packages/open-payments/src/client/outgoing-payment.test.ts#L148-L150

Reply:
Right, we usually don't have those validator tests for routes/handlers that directly call `get` or `post`

Reply:
The exported rotate fucntion should look similar to
https://github.com/interledger/rafiki/blob/1124707caf2f6ed9f7ddae777898973f9107b749/packages/open-payments/src/client/outgoing-payment.ts#L79-L91

Reply:
@wilsonianb we could probably test like I was doing in the other PR https://github.com/interledger/rafiki/pull/848/files#diff-4f318c2a374cc145f8825fecf95d77b29a728927b5a64ce393897b01becb2b93R95-R118